numpyro/distributions/transforms.py

```diff
@@ -282,7 +282,7 @@ def __eq__(self, other):
 
 def _get_compose_transform_input_event_dim(parts):
     input_event_dim = parts[-1].domain.event_dim
-    for part in parts[len(parts) - 1 :: -1]:
+    for part in parts[len(parts) - 1 : len(parts) : -1]:
```

Comment:
Thanks for the catch! I think we need to ignore the -1 part here. Maybe using `parts[:-1][::-1]` to be less confusing?

Reply:
It was surprised to me that
```
x = [10]
print(x[-2::-1])  # Output: [10] 
```

Reply:
Sounds great!